lib/mail/field.rb

```diff
@@ -232,12 +232,13 @@ def split(raw_field)
     #  that is immediately followed by WSP.  Each header field should be
     #  treated in its unfolded form for further syntactic and semantic
     #  evaluation.
-    def unfold(string)
-      string.gsub(/[\r\n \t]+/m, ' ')
+    def unfold(name, value)
+      regex = name == 'Subject' ? /\s*\r\n\s+/m : /[\r\n \t]+/m
```

Comment:
Which fields? Is there a chance the specs are wrong too and written just to
match the existing behavior?

On April 5, 2016 at 6:57:41 PM, andrew selder (notifications@github.com)
wrote:

> In lib/mail/field.rb
> <https://github.com/mikel/mail/pull/981#discussion_r58633946>:
>
> > @@ -232,12 +232,13 @@ def split(raw_field)
> >      #  that is immediately followed by WSP.  Each header field should be
> >      #  treated in its unfolded form for further syntactic and semantic
> >      #  evaluation.
> > -    def unfold(string)
> > -      string.gsub(/[\r\n \t]+/m, ' ')
> > +    def unfold(name, value)
> > +      regex = name == 'Subject' ? /\s*\r\n\s+/m : /[\r\n \t]+/m
>
> @kjg <https://github.com/kjg> That was my original approach, but based on
> existing specs, there are some header fields where the whitespace is
> supposed to be stripped out. I wasn't able to come up with a single regex
> that passed both my new specs and all existing specs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mikel/mail/pull/981/files/c93e25218db01b6177a34323b58bde30684ce57b#r58633946>
>